opentech/apply/utils/views.py

```diff
@@ -169,3 +173,66 @@ def post(self, request, *args, **kwargs):
             self.object = None
 
         return super().post(request, *args, **kwargs)
+
+
+class RedirectToListViewMixin:
+    destination_lut = None
+
+    def get(self, request, *args, **kwargs):
+        if not request.GET:
+            return super().get(request, *args, **kwargs)
+
+        destination_lut = self.get_destination_lut()
+
+        response = self.redirect_to_list_view(request.GET, destination_lut)
+        if response is None:
+            return super().get(request, *args, **kwargs)
+
+        return response
+
+    def get_destination_lut(self):
+        if self.destination_lut is None:
+            raise ImproperlyConfigured('destination_lut must be set to use RedirectToListViewMixin')
+
+        if not isinstance(self.destination_lut, dict):
+            raise ImproperlyConfigured('destination_lut should be a dictionary')
+
+        return self.destination_lut
+
+    def redirect_to_list_view(self, query_dict, destination_lut):
+        """
+        Attempt to redirect to a list view when filtering a table
+
+        Each table is created with a prefix which is used when sorting keys are
+        generated.  We use those keys here to decide where the user should be
+        redirected to.
+        """
+        first_key, sep, _ = list(query_dict.keys())[0].rpartition('--')
+        prefix = f'{first_key}{sep}'
+
+        if prefix not in destination_lut.keys():
+            # the prefix we found wasn't in the expected list so exit now
+            return
+
+        path = destination_lut[prefix]
+
+        query_args = dict(self.strip_prefix(query_dict, prefix))
```

Comment:
do we need to drop querystrings that do not have the prefix?

Reply:
Again I think this depends wholly on the outcome of [this discussion](https://github.com/OpenTechFund/opentech.fund/pull/1477#discussion_r320667734).